### PR TITLE
[MIG] stock_putaway_hook: Migration to 16.0

### PR DESCRIPTION
**Pull request by rousseldenis**
Code has been changed in Odoo core: the _get_putaway_strategy() should always return a location.

This has been changed here.

### CODE REVIEW

**Review by baimont — APPROVED** [review]

**Comment by rousseldenis** [comment]
Waiting a little bit, I'm adding tests

**Comment by rousseldenis** [comment]
This is ready

**Comment by baimont** [comment]
> Waiting a little bit, I'm adding tests

a nice idea

File: stock_putaway_hook/tests/test_putaway_hook.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+from odoo.tests.common import TransactionCase
+
+from .common import PutawayHookCommon
+
+
+class TestPutawayHook(PutawayHookCommon, TransactionCase):
+    def test_putaway_hook(self):
+        """
+        The first location has no strategy nor alternative one
+        The _get_putaway_strategy() should returns that location
```
Comment (baimont) [inline stock_putaway_hook/tests/test_putaway_hook.py]: should return

File: stock_putaway_hook/tests/test_putaway_hook.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        self.assertEqual(
+            self.location_internal_1,
+            strategy,
+        )
+
+    def test_putaway_hook_context(self):
+        """
+        The second location has no strategy nor alternative one
+        But _putaway_<field> is passed in context.
+        Should returns the second location
```
Comment (baimont) [inline stock_putaway_hook/tests/test_putaway_hook.py]: should return

File: stock_putaway_hook/tests/test_putaway_hook.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        )._get_putaway_strategy(self.product)
+        self.assertEqual(
+            self.location_internal_1,
+            strategy,
+        )
+
+    def test_putaway_hook_alternative(self):
+        """
+        The third location has no strategy but a foo alternative one
+        Should returns the location from the putaway
```
Comment (baimont) [inline stock_putaway_hook/tests/test_putaway_hook.py]: should return

**Review by sebalix — APPROVED** [review]

**Pull request merged**